### PR TITLE
added ImportedFile model/table and refactored import_from_agave

### PR DESCRIPTION
**Pull request by reptillicus**
I added another commit that changes the way that feature assets can be added, done directly from tapis synchronously.

### CODE REVIEW

File: geoapi/tasks/external_data.py [hunk 1]
```diff
@@ -118,6 +114,11 @@ def import_from_agave(user: User, systemId: str, path: str, proj: Project):
                     tmpFile.close()
                 else:
                     continue
+                # Save the row in the database that marks this file as already imported so it doesn't get added again
+                targetFile = ImportsService.createImportedFile(proj.id, systemId, str(item.path), item.lastModified)
```
Comment (nathanfranklin) [inline geoapi/tasks/external_data.py:118]: I wonder how we should handle when users delete a feature on an observable project.

I wonder if we should treat observable projects differently.  Maybe, we should display it differently in the frontend and disable some things for this map (feature deletion in the front end and in the back end, file adding in the front end and in the backend).  We would convey that the map is a one-to-one mapping of the directory and that users have to make direct changes (like adding or deleting files) to the DS directory.

Comment (nathanfranklin) [reply geoapi/tasks/external_data.py:118]: If a project is deleted, we should probably delete the ImportedFile rows that are related to that project.  Should we just add a ticket for now?

File: geoapi/models/imported_file.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+class ImportedFile(Base):
+    __tablename__ = 'imported_file'
+
+    id = Column(Integer, primary_key=True)
+    project_id = Column(Integer, ForeignKey('projects.id', ondelete="CASCADE"), index=True)
+    system_id = Column(String, nullable=False, index=True)
+    path = Column(String, nullable=False, index=True)
+    # last_updated is the timestamp on the file itself as to when it was last touched
+    last_updated = Column(DateTime(timezone=True), nullable=False)
```
Comment (nathanfranklin) [inline geoapi/models/imported_file.py:16]: should we add a relationship to what feature(s) correspond to this file

File: geoapi/tasks/external_data.py (hunk 1 above)
Comment (reptillicus) [reply geoapi/tasks/external_data.py]: > If a project is deleted, we should probably delete the ImportedFile rows that are related to that project. Should we just add a ticket for now?

The ondelete="CASCADE" on the project_id should handle it

Comment (reptillicus) [reply geoapi/tasks/external_data.py]: Yeah we should start thinking of how to do "published" projects that are read only. How much time would it take to implement what you mentioned though? I like the idea, I'm just lazy.

File: geoapi/models/imported_file.py (hunk 2 above)
Comment (reptillicus) [reply geoapi/models/imported_file.py:16]: I actually started doing that but it started to get complicated. I think would have to do a many-to-many table bubt I couldn't figure out a good place to put that logic. Thoughts?

File: geoapi/tasks/external_data.py (hunk 1 above)
Comment (nathanfranklin) [reply geoapi/tasks/external_data.py]: Making a list to send you via slack to discuss

**Pull request merged**